cachito/web/validation.py

```diff
@@ -0,0 +1,48 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from typing import Any, Dict, List
+
+from cachito.errors import ValidationError
+
+
+def validate_replacement(replacement: Dict[str, Any]) -> None:
+    """
+    Validate the JSON representation of a dependency replacement.
+
+    :param replacement: the JSON representation of a dependency replacement
+    :type replacement: dict[str, any]
+    :raise ValidationError: if the JSON does not match the required schema
+    """
+    required = {"name", "type", "version"}
+    optional = {"new_name"}
+
+    if not isinstance(replacement, dict) or (replacement.keys() - required - optional):
+        raise ValidationError(
+            "A dependency replacement must be a JSON object with the following "
+            f'keys: {", ".join(sorted(required))}. It may also contain the following optional '
+            f'keys: {", ".join(sorted(optional))}.'
+        )
+
+    for key in required | optional:
+        # Skip the validation of optional keys that are not set
+        if key not in replacement and key in optional:
+            continue
+
+        if not isinstance(replacement[key], str):
+            raise ValidationError(
+                'The "{}" key of the dependency replacement must be a string'.format(key)
+            )
+
+
+def validate_dependency_replacements(replacements: List[Dict[str, Any]]) -> None:
```

Comment:
Do we have a test for this?

Reply:
`test_create_request_invalid_dependency_replacement` is restored.

Reply:
And other existing test like `test_create_and_fetch_request` covers the valid replacements.

Reply:
Gotcha, seems reasonable :+1: